app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -333,7 +333,30 @@ private void runInitSequence(Boolean loadCalibrationData) throws IOException {
                             }
                         }
 
-                        // todo : port remaining function
+                        ArrayUtils.reverse(fitVals);
+                        PolynomialFunction fitFunction = new PolynomialFunction(fitVals);
+                        double[] yData = new double[DACX.length];
+                        for (int j = 0; j < DACX.length; j++) {
+                            yData[j] = fitFunction.value(DACX[j]) - (OFF.get(j) * slope + intercept);
+                        }
+                        int LOOKBEHIND = 100;
+                        int LOOKAHEAD = 100;
```

Comment:
both values can be assigned a single variable. But i'm not sure that the algorithm is the most efficient.

The basic goal of the algorithm is to locate the DAC value that will yield a voltage closest to the requested one. 
![dac error](https://cloud.githubusercontent.com/assets/6078658/26584832/dbb0e80e-4567-11e7-96ae-09e00ca9c46d.png)
As you can see, digital to analog convertors deviate from the theoretical setpoint in an irregular manner. The PSLab stores the error present in each DAC code in an array of 4K elements (12-bit = 4096 codes) .
The object of the algorithm is to search for codes in the vicinity of the requested code in order to yield an output that is closest to the requested setpoint.   And as is evident from the yellow plot, this approach works fine and manages to get full 12-bit performance from the DAC. 

But because the calibration process is so long, we should probably switch to a quadratic polynomial correction for each DAC channel. The maximum error one can expect after quadratic correction is not more than 5mV .